### PR TITLE
Don't let .netrc take precedence in ApiClient

### PR DESCRIPTION
**Pull request by kartikgupta-db**
- Python requests library automatically picks up `.netrc` configurations for authentication. (https://github.com/psf/requests/issues/2773)
- This PR disables this behaviour to make it consistent with the docs (https://docs.databricks.com/dev-tools/cli/index.html#set-up-authentication) 
- The fix uses a workaround which should be stable until the API for authentication in requests library changes.

### CODE REVIEW

**Review by fjakobs — COMMENTED** [review]
Fix looks reasonable but I'd like to get @pietern's opinion before merging since it would be a breaking change.

**Comment by pietern** [comment]
We can't merge this as is. Reading through the linked issues, there's an opportunity to do this in a non-breaking way. The default auth handler checks the host and overrides a token from `.netrc` if present, regardless of existing authentication being configured. What we can do instead is fallback to reading from `.netrc` if nothing is configured. Then the case where a user has a `.netrc` and only specifies a hostname, the CLI will continue to work.

**Comment by pietern** [comment]
This is a nice workaround. In the body we could check for the `Authorization` header and only defer to `.netrc` if it isn't set.

**Comment by kartikgupta-db** [comment]
> We can't merge this as is. Reading through the linked issues, there's an opportunity to do this in a non-breaking way. The default auth handler checks the host and overrides a token from `.netrc` if present, regardless of existing authentication being configured. What we can do instead is fallback to reading from `.netrc` if nothing is configured. Then the case where a user has a `.netrc` and only specifies a hostname, the CLI will continue to work.

The cli checks that there should be atleast 1 form of accepted configuration methods (spark config, environment variables or databrickscfg) are present. So there should be no cli user directly dependent on a `.netrc`.

For users using the api client directly from the sdk, the fallback should provide them the similar experience as they have now, with the caveat that now other forms of configurations will take precedence (earlier `.netrc` had the highest precedence).  But this is an undocumented use case.

File: databricks_cli/sdk/api_client.py
Comment (pietern) [inline databricks_cli/sdk/api_client.py:76]: This second example isn't valid Python.

**Review by pietern — COMMENTED** [review]
Please add a couple tests:

* `.netrc` exists with a valid entry, but auth is specified elsewhere.
* `.netrc` exists with a valid entry, auth is not specified.
* `.netrc` exists with an invalid entry
* `.netrc` does not exist

As is, I'm worried what would happen all cases but the first.

**Comment by pietern** [comment]
> For users using the api client directly from the sdk, the fallback should provide them the similar experience as they have now, with the caveat that now other forms of configurations will take precedence (earlier .netrc had the highest precedence). But this is an undocumented use case.

It's one we need to call out in the changelog / release notes though ;-)

**Review by pietern — COMMENTED** [review]
What happens if you have a netrc file with garbage / invalid content?

I expect the default behavior in the requests library is to ignore it. Can you confirm this implementation does the same?

**Review by pietern — APPROVED** [review]
Looks great, thank you!

**Pull request merged**
